### ISSUE TITLE
@internal marked classes is inaccessible from another namespaces

### ISSUE THREAD

**Issue by russelomua** [getsentry/sentry-php#1655]
### How do you use Sentry?

Sentry SaaS (sentry.io)

### SDK version

4.0+

### Steps to reproduce

Creating client with `ClientBuilder::create([])` is failing psalm inspection with error [InternalClass](https://psalm.dev/174).

Also  try to create implementations by myself and is fail too because `PayloadSerializer`, `HttpClient` and `HttpTransport` also maked as `@internal`

```php
$options = new Options([]);
$client = new Client(
    options: $options,
    transport: new HttpTransport(
        options: $options,
        httpClient: new HttpClient(Client::SDK_IDENTIFIER, CLient::SDK_VERSION),
        payloadSerializer: new PayloadSerializer($options),
        logger: $logger,
    ),
    logger: $logger,
);
$hub = new Hub($client);
```

Please give an alternative way to create a client for framework agnostic or custom frameworks projects without using singleton pattern via`\Sentry\init([])`

### Expected result

Psaml inspection completes without errors

### Actual result

```cli
ERROR: InternalClass - {projectClass}.php - Sentry\ClientBuilder is internal to Sentry but called from {projectClass} (see https://psalm.dev/174)
```

**Comment by cleptric** [comment]
You can ignore these in your local psalm config https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#suppressing-issues.

**Comment by cleptric** [comment]
https://github.com/getsentry/sentry-php/pull/1661

**Comment by cleptric** [comment]
Shipped in [4.2.0](https://github.com/getsentry/sentry-php/releases/tag/4.2.0)
